### PR TITLE
keep mergify and github branch protection in sync

### PR DESCRIPTION
**Pull request by cmr**
This adds .mergify.yml to the `test.py` `render` step, and a script that informs the github API of the new required status checks only on master.

### CODE REVIEW

File: .mergify.yml.jinja [hunk 1]
```diff
@@ -0,0 +1,13 @@
+pull_request_rules:
+    - name: automatically merge approved PRs with the ready-to-merge label
+      conditions:
+          {%- for status in required_status %}
+          - "status-success={{status}}"
+          {%- endfor %}
+          - "#approved-reviews-by>=1"
+          - label=ready-to-merge
```
Comment (nholland94) [inline .mergify.yml.jinja]: Will mergify still respect the github required checks? Or if all these are met, will it just merge anyway?

File: scripts/test.py [hunk 2]
```diff
@@ -177,15 +199,16 @@ def make_log_dir():
 
     print('Testing successfull')
 
+
 def render(args):
```
Comment (nholland94) [inline scripts/test.py]: This function is a little long now, but it's just a script so it's fine.

File: scripts/test.py [hunk 3]
```diff
@@ -197,10 +220,45 @@ def render(args):
...
+
+    rendered = template.render(
+        required_status=required_status
+    )
+
+    if args.check:
+        with open(output_file, 'r') as file:
+            if file.read() != rendered:
+                fail('mergify configuration is out of date, re-render it')
+    elif not args.required:
```
Comment (nholland94) [inline scripts/test.py]: It's not entirely clear to me why there is this special case logic for required. Couldn't you just pass in `--check` to not write the files and then `--required` will only print out the extra info you want? Or you could even put `required` as it's own top level command.

File: .mergify.yml.jinja (hunk 1 above)
Comment (cmr) [reply .mergify.yml.jinja]: Mergify has never respected the github required checks

File: scripts/test.py (hunk 3 above)
Comment (cmr) [reply scripts/test.py]: With the current logic, `--required` will always print out the required status checks. If we used `--check`, the script might fail with no effect if somehow master got a stale `.mergify.yml`/`.circleci/config.yml`.

Comment (cmr) [reply scripts/test.py]: It should just be a top-level command.

**Pull request merged**